app/views/sign_up/passwords/new.html.erb

```diff
@@ -11,12 +11,13 @@
       method: :post,
       html: { autocomplete: 'off' },
     ) do |f| %>
-    <%= render PasswordConfirmationComponent.new(
-          form: f,
-          field_options: {
-            input_html: { aria: { describedby: 'password-description' } },
-          },
-        ) %>
+  <%= text_field_tag('username', @email_address.email, hidden: true, autocomplete: 'username') %>
```

Comment:
Testing this in the real-world, it works well, and [Chromium's guidance](https://www.chromium.org/developers/design-documents/form-styles-that-chromium-understands/) suggests that a visually-hidden `[type=text]` is the recommended approach, but I'm a bit concerned if a browser may try to auto-fill this field, even if visually hidden with the `[hidden]` attribute. I could imagine some strange scenarios if the user already had a saved credential with the site, if autofill could cause the user to be prompted to save that password for the wrong email address.

In reviewing [the relevant specification](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-expectation-mantle), it seems like the specification is explicit that the behavior we want should be applied through `[type=hidden]`:

>There are two ways this [`autocomplete`] attribute is used. When wearing the autofill expectation mantle, the [autocomplete](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete) attribute describes what input is expected from users. When wearing the autofill anchor mantle, the [autocomplete](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete) attribute describes the meaning of the given value.
>
>On an [input](https://html.spec.whatwg.org/multipage/input.html#the-input-element) element whose [type](https://html.spec.whatwg.org/multipage/input.html#attr-input-type) attribute is in the [Hidden](https://html.spec.whatwg.org/multipage/input.html#hidden-state-(type=hidden)) state, the [autocomplete](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete) attribute wears the [autofill anchor mantle](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-anchor-mantle). In all other cases, it wears the [autofill expectation mantle](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-expectation-mantle).

tl;dr: `[type=hidden]` = "autofill anchor mantle" = "describes the meaning of the given value" (as opposed to "describes what input is expected from users")

I might try this out locally to see if it works the same to change `[type=text]` to `[type=hidden]`, since it seems a little safer / closer to the spec..

Reply:
Changing from `[type=text]` to `[type=hidden]`, it still worked as expected in Safari, but it regressed to the original behavior in Firefox and Chrome.

I think this is the implementation we'll want to keep. For extra assurance, I tested autofill with a different credential's password in Chrome, Safari, and Firefox with `[type=text][hidden]`, and none of them manipulated the value.